elytron/ELY-1712-enhanced-audit-logging.adoc

```diff
@@ -0,0 +1,283 @@
+= ELY-1712 Enhanced Audit Logging
+:author:            Justin Michael Cook
+:email:             jucook@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+:issue-base-url:    https://issues.jboss.org/browse
+
+== Overview
+
+Currently, Audit Logging in Elytron does not have the same capabilities as Legacy Security. This task is to enhance Elytron Audit Logging to have the same capabilities as Legacy Security Audit Logging.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/ELY-1712[ELY-1712]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-699[EAP7-699]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:jkasik@redhat.com[Jan Kašík]
+
+=== Testing By
+
+[X] Engineering
+
+[ ] QE
+
+=== Affected Projects or Components
+
+* Security
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+
+* Support for RFC 5424 should be enhanced, according to the following parameters, all of which should be added to the ```syslogAuditLogType``` in the Elytron schema
+// Could not find information about this in legacy
+** A ```syslog-format``` attribute should be added which supports a value of ```RFC5424```
+** Some of the parameters needed for RFC 5424 already exist in the Elytron codebase, which are the following:
+*** severity
+*** server address
+*** port
+*** setting TCP
+*** setting SSL
+*** setting socket factory
+*** setting hostname
+** The current attributes available in the Elytron schema, which will be used for RFC 5424, are:
+*** server-address
+*** port
+*** transport type
+*** message format
+*** hostname (?)
+*** ssl-context
+** Additional attributes which will need to be added are:
```

Comment:
Please include type of attributes which will be added (`String`, `int`...).

Reply:
Some of the attributes I won't know what class they would be until I start implementing it, but I'll define the ones I can